docs/source/metadata_source_spec.rst

```diff
@@ -0,0 +1,138 @@
+Metadata source specification
+*****************************
+
+This metadata source specification defines how to structure a collection of metadata records
+that together form the source material for a ``datalad-catalog`` catalog instance.
+
+The specification benefits both users and developers in that it separates metadata formats
+from the tooling that processes it:
+
+- users can create and maintain such specification-compliant metadata collections without
+  having to employ ``datalad-catalog`` tooling
+- both generic and format-specific tooling can be developed and deployed, either as part of
+  ``datalad-catalog`` or as custom extensions, to transform specification-compliant metadata
+  collections into a state renderable by a catalog
+
+
+High-level design
+=================
+
+The metadata source specification supports:
+
+1. **Per-catalog versioned customizations**: the top-level functional unit of the source
+   specification is a catalog instance, which can be customized via a versioned configuration
+   file as defined in the section :doc:`catalog_config`. This means a specification-compliant
+   collection of records can specify the (version-specific) "look and feel" of a catalog,
+   in addition to its displayed content.
+2. **Multi-dataset, multi-version records**: the source specification has a filesystem layout
+   with a directory for each unique dataset identifier, which in turn has a subdirectory for
+   each unique version identifier of a given dataset. This ensures a modular setup within which
+   records for multiple versions of the same dataset can coexist.
+3. **Multi-format metadata records**: the specification places no restrictions on the number
+   and type of metadata records in a collection for a given dataset version, since in reality
+   metadata often originate from a variety of sources and exist in a variety of formats.
+   The transformation of different record formats into ``datalad-catalog``-compatible records
+   is conveniently shifted into the tooling domain, and is not part of the specification itself.
+
+
+The specification
+=================
+
+The following filesystem layout and record naming scheme should be adhered to for
+a given collection of records:
+
+.. code-block::
+
+   .
+   ├── config/
+   │   └── <config-version-id>/
+   │       └── config.json
```

Comment:
Another point about the config, it can also include a logo path (specified relative to the location of the config, within the context of the environment running the `datalad-catalog` code). For the purposes of the collection, this logo will either have to be provided as an image file in the collection itself (likely alongside the `config.json` file) or as a downloadable URL. Thoughts?